cmplxs/cmplxs.go

```diff
@@ -0,0 +1,669 @@
+	}
+	return math.Pow(norm, 1/L)
+}
+
+// Div performs element-wise division dst / s
+// and stores the value in dst. It panics if the
+// lengths of s and t are not equal.
+func Div(dst, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	c128.Div(dst, s)
+}
+
+// DivTo performs element-wise division s / t
+// and stores the value in dst. It panics if the
+// lengths of s, t, and dst are not equal.
+func DivTo(dst, s, t []complex128) []complex128 {
+	if len(s) != len(t) || len(dst) != len(t) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	return c128.DivTo(dst, s, t)
+}
+
+// Dot computes the dot product of s1 and s2, i.e.
+// sum_{i = 1}^N s1[i]*s2[i].
+// A panic will occur if lengths of arguments do not match.
+func Dot(s1, s2 []complex128) complex128 {
+	if len(s1) != len(s2) {
+		panic("cmplxs: lengths of the slices do not match")
+	}
+	return c128.DotUnitary(s1, s2)
+}
+
+// Equal returns true if the slices have equal lengths and
+// all elements are numerically identical.
+func Equal(s1, s2 []complex128) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, val := range s1 {
+		if s2[i] != val {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualApprox returns true if the slices have equal lengths and
+// all element pairs have an absolute tolerance less than tol or a
+// relative tolerance less than tol.
+func EqualApprox(s1, s2 []complex128, tol float64) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, a := range s1 {
+		if !EqualWithinAbsOrRel(a, s2[i], tol, tol) {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualFunc returns true if the slices have the same lengths
+// and the function returns true for all element pairs.
+func EqualFunc(s1, s2 []complex128, f func(complex128, complex128) bool) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i, val := range s1 {
+		if !f(val, s2[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualWithinAbs returns true if parts of a and b have an absolute
+// difference of less than tol.
+func EqualWithinAbs(a, b complex128, tol float64) bool {
+	return a == b || cmplx.Abs(a-b) <= tol
+}
+
+// EqualWithinRel returns true if the difference between parts of a and b
+// is not greater than tol times the greater value.
+func EqualWithinRel(a, b complex128, tol float64) bool {
+	if a == b {
+		return true
+	}
+	return floats.EqualWithinRel(real(a), real(b), tol) && floats.EqualWithinRel(imag(a), imag(b), tol)
+}
+
+// EqualWithinAbsOrRel returns true if parts of a and b are equal to within
+// the absolute tolerance.
+func EqualWithinAbsOrRel(a, b complex128, absTol, relTol float64) bool {
+	if EqualWithinAbs(a, b, absTol) {
+		return true
+	}
+	return EqualWithinRel(a, b, relTol)
+}
+
+// EqualWithinULP returns true if parts of a and b are equal to within
+// the specified number of floating point units in the last place.
+func EqualWithinULP(a, b complex128, ulp uint) bool {
```

Comment:
I don't know how to make this consistent with the other `Equal` functions and I'm also not sure about the utility/necessity of this.

Reply:
I can drop it. I don't see it used often in floats.

Reply:
Yes, I would drop it.

Reply:
Done